### PR TITLE
Start plumbing in Descriptors

### PR DESCRIPTION
**Pull request by ZackButcher**
This PR adds a `DescriptorFinder` interface and implementation, and adds it to the config manager. Subsequent PRs will add it as a parameter to the aspect's `ValidateConfig` methods. It's likely I'll need to restructure how/where the descriptor finder is created and passed around in the config validator and manager as a result of those subsequent PRs, I only touched them in this PR because our linters complained about dead code if I didn't create a descriptor finder somewhere outside of a test.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/369)

### CODE REVIEW

File: pkg/config/descriptors/descriptors.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import (
+	dpb "istio.io/api/mixer/v1/config/descriptor"
+	pb "istio.io/mixer/pkg/config/proto"
+)
+
+// Finder describes anything that can provide a view into the config's descriptors by name and type.
+type Finder interface {
+	// GetLog retrieves the log descriptor named `name`
+	GetLog(name string) (*dpb.LogEntryDescriptor, bool)
```
Comment (douglas-reid) [inline pkg/config/descriptors/descriptors.go]: should these be named `Find*` to match `Finder` interface?  or should we name this interface `Getter` ?

Comment (mandarjog) [inline pkg/config/descriptors/descriptors.go]: `Finder` and `Find*` is better.

File: pkg/config/descriptors/descriptors.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+	// GetMetric retrieves the metric descriptor named `name`
+	GetMetric(name string) (*dpb.MetricDescriptor, bool)
```
Comment (mandarjog) [inline pkg/config/descriptors/descriptors.go]: When the return type is a `pointer` 2nd `bool` return is redundant. 
you can always test the pointer for `nil`

File: pkg/config/descriptors/descriptors.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		logs:               logs,
+		metrics:            metrics,
+		monitoredResources: monitoredResources,
+		principals:         principals,
+		quotas:             quotas,
+	}
+}
+
+func (d finder) GetLog(name string) (*dpb.LogEntryDescriptor, bool) {
+	l, found := d.logs[name]
```
Comment (mandarjog) [inline pkg/config/descriptors/descriptors.go]: it simplifies all this code, just return the pointer.

**Pull request merged**